### PR TITLE
fix: resolve ios focus/blur issue

### PR DESCRIPTION
**Pull request by aMollusk**
**What**:  Resolving #421 

**Why**: To fix the iOS issue where the the blur event is triggered after the on click. 

**How**: Adding a `setTimeout` to the `toggleMenu` call inside the click handler.

**Checklist**:

* [ ] Documentation N/A
* [x] Tests
* [x] Ready to be merged
      
* [x] Added myself to contributors table

### CODE REVIEW

**Comment by arahansen** [comment]
Before I open a ticket to discuss an issue I'm seeing, I wanted to maybe ask about it here:

with this change, my tests for components consuming Downshift are broken as I now need to apply fake timers any time I interact with my component. This makes writing tests for components that consume Downshift less intuitive to write. Is there possibly another way around this that would keep Downshift more synchronous?

Also possible: perhaps I should be updating my tests to make downshift components easier to test?

**Comment by kentcdodds** [comment]
Sorry that this broke your tests @arahansen. If you can think of a way to make this synchronous without also having the problems this PR was intended to solve that would be super.

Alternatively, have you looked into [`react-testing-library`](https://github.com/kentcdodds/react-testing-library)? It exposes `wait` and `waitForElement` which could make things a little more straightforward and less tied to implementation details for your downshift tests.

**Comment by arahansen** [comment]
The `waitForElement` sounds like a terrific idea! My team and I have looked into react-testing-library. I think our biggest concern is around the rewrite that would be required to port our tests from enzyme to react-testing-library (if you have any thoughts on that, I'd love to hear them!) 

But in the meantime, I might try pulling in an implementation of waitForElement that can work alongside enzyme

File: src/__tests__/downshift.get-button-props.js [hunk 1]
```diff
@@ -19,10 +19,12 @@ test('space on button opens and closes the menu', () => {
 test('clicking on the button opens and closes the menu', () => {
   const {button, renderSpy} = setup()
   Simulate.click(button)
+  jest.runAllTimers()
```
Comment (kentcdodds) [inline src/__tests__/downshift.get-button-props.js]: Turns out this is required by pretty much anyone testing a component that uses downshift and it's a bit annoying. @aMollusk, could you jump into https://github.com/paypal/downshift/issues/437 and chime in to discuss an alternative/workaround for tests?

**Pull request merged**